msrestazure/azure_active_directory.py

```diff
@@ -544,8 +544,7 @@ def signed_session(self):
             raw_token = self._adal_method(*self._args, **self._kwargs)
         except adal.AdalError as err:
             # pylint: disable=no-member
-            if (hasattr(err, 'error_response') and ('error_description' in err.error_response)
-                    and ('AADSTS70008:' in err.error_response['error_description'])):
+            if 'AADSTS70008:' in ((getattr(err, 'error_response', None) or {}).get('error_description') or ''):
```

Comment:
Why not `getattr(err, 'error_response', {}).get('error_description', "")` ? that's the point of the last parameter to be the default, I find it more Pythonic,

Reply:
The problem with that is if you have the parameter, but it's not what you expect, you could still get an error.
For instance, say that `err.error_response == None`. In this case, `getattr(err, 'error_response', {}) == None`, and `get` would throw an exception. Same goes for the dict.

Reply:
gotcha, good point